packages/@ember/-internals/routing/lib/services/router.ts

```diff
@@ -393,6 +394,32 @@ export default class RouterService extends Service {
     return this._router._routerMicrolib.recognizeAndLoad(internalURL);
   }
 
+  /**
+   * Refreshes all currently active routes, doing a full transition.
+   * If a routeName is provided and refers to a currently active route,
+   * it will refresh only that route and its descendents.
+   * Returns a promise that will be resolved once the refresh is complete.
+   * All resetController, beforeModel, model, afterModel, redirect, and setupController
+   * hooks will be called again. You will get new data from the model hook.
+   *
+   * @method refresh
+   * @param {String} [routeName] the route to refresh (along with all child routes)
+   * @return Transition
+   * @public
+   */
+  refresh(routeName?: string) {
+    if (!routeName) {
+      return this._router._routerMicrolib.refresh();
+    }
+
+    assert(`The route "${routeName}" was not found`, this._router.hasRoute(routeName));
```

Comment:
I'm not sure if these are clear enough.I used the same assert as in the [router](https://github.com/emberjs/ember.js/blob/d414901668464b86fa96e2c724a8ce907d2b064d/packages/%40ember/-internals/routing/lib/system/router.ts#L878) for the first one.